docs/Processes.md

```diff
@@ -555,6 +555,372 @@ The modifications from the standard tick.q include:
+
+- defaultbatch:
+
+  This is effectively the standard TP batching mode where, upon receiving a tick, the STP immediately logs it to disk and batches the update which is published to subscribers whenever the timer function is next called. This mode represents a good balance of latency and overall throughput.
+
+- immediate:
+
+  In this mode no batching occurs, and the update is logged and published immediately upon entering the STP. This is less efficient in terms of overall throughput but ensures low latency.
+
+- memorybatch:
+
+  In this mode, neither logging nor publishing happens immediately but everything is held in memory until the timer function is called, at which point the update is logged and published. High overall message throughput is possible with this mode, but there is a risk that some messages aren't logged in the case of STP failure. Also note that the the ordering of messages from different tables in the log file will not align with arrival order.
+
+### Starting a Segmented Tickerplant process
+
+Starting an STP process is similar to starting a tickerplant, we need to have an updated process.csv that contains a line for the STP process like the one below. Optional flags such as `-.stplg.batchmode` and `-.stplg.errmode` can be added to change settings for the process.
+
+```
+localhost,${KDBBASEPORT}+103,segmentedtickerplant,stp1,${TORQAPPHOME}/appconfig/passwords/accesslist.txt,1,0,,,${KDBCODE}/processes/segmentedtickerplant.q,1,-schemafile ${TORQAPPHOME}/database.q -.stplg.batchmode immediate -.stplg.errmode 0 -t 1,q
+```
+
+The process can either be started using:
+
+```shell
+bash torq.sh start stp1
+```
+
+or:
+
+```shell
+q ${TORQHOME}/torq.q -proctype segmentedtickerplant -procname stp1 -load ${KDBCODE}/processes/segmentedtickerplant.q
+```
+
+### Subscriptions
+
+Subscribing to the STP works in a very similar fashion to the original tickerplant. From the subscriber's perspective the subscription logic is backwardly compatible: it opens a handle to the STP and calls `.u.sub` with a list of tables to subscribe to as its first argument and either a null symbol or a list of symbols as a sym filter.  The STP also supports a keyed table of conditions (in q parse format) and a list of columns that should be published. 
+
+Whilst complex bespoke subscription is possible in the STP it is generally not recommended. Complex subscription filtering should be off loaded to a chained STP.
+
+```q
+// Subscribe to everything
+handletoSTP(`.u.sub;`;`)
+
+// Subscribe GOOG and AAPL symbols in the trade table
+handletoSTP(`.u.sub;`trade;`GOOG`AAPL)
+
+// Subscribe to all tables but with custom conditions
+handletoSTP(`.u.sub;`;conditions)
+...
+q) show conditions
+tabname| filts            columns
+-------| --------------------------------
+trade  | `                `time`sym`price
+quote  | `bid<200`bid>100 `
```

Comment:
IS this filts condition correct? Can you talk to me about this, I don't understand it.